ajax/ajax.md

```diff
@@ -16,74 +17,94 @@ About 3 hours
-- Video walkthrough of lesson slides [AJAX](https://drive.google.com/file/d/1pLyMqvC-8cFroVTWMUBeG1_BRd4L-rL3/view?usp=sharing)
+- Read through lesson slides [AJAX](https://drive.google.com/file/d/12KSxnLTCCRl2wG3G3cPjv_9ZF-mltcKx/view?usp=sharing)
+
 
+### What is AJAX ?
+AJAX stands for Asynchronous JavaScript and XML. It is a **technique** by which front-end get data from the backend by making *asynchronous* API calls.
+Because AJAX is *asynchronous*, you must pass in a callback function to handle the received data. Making an AJAX request without any callback function will not have any effect on your application.
 
-### What is AJAX?
+### Full Load Page V/s AJAX
+> Lets say there is an input box and a button. Onclick of button the page shows some data say *hello world* send by the *server*. When the button is clicked -  
+#### Full load Application Flow
+1. The browser sends a request to a server designated by the URL. 
+2. The server does some processing, probably queries a database, then responds to the browser by sending back HTML and JavaScript
+3. The browser renders that HTML and JavaScript into a webpage.
 
-AJAX stands for Asynchronous JavaScript and XML. It is a way for front-end code to talk to the backend by making *asynchronous* API calls.
+#### AJAX Flow
+- The client sends a request to the server.
+- The server does some processing and returns the corresponding data (usually in JSON format).
+- When the request gets completed the callback function is executed in which data is present. That data is shown at frontend in any way you want.
 
-What is *asynchronous*?  You can think of this as a sequence of activities that respond to one another, but not in a continuous stream of time. Synchronous activities happen in a continuous, uninterrupted stream of time. A phone call, for example, is a synchronous method of communication, because two people are on the phone at the same time having a conversation. Email, on the other hand, is asynchronous communication. Even though a single conversation will happen across a thread of emails, each email by itself has a gap of time after the preceding email. On top of this, you never know when the next email response will arrive. This is the same thing that happens with asynchronous server calls. The front end will send a request to the backend, there is an indeterminable period of time that passes, then the server sends a response.
+| Difference Point        | Full Load Page          | AJAX  |
+| ------------- |:-------------| -----:|
+| Client requests to server   | waits for response  | can perform other task simultaneously  |
+| Server Response       | responds a corresponding HTML page  |  responds with data (preferably JSON ) |
+| Render |  a complete new page is rendered   | only that part of page changes which has requested |
+| Thread Blocked | true   | false |
+| Data Received by | the browser in form of HTML   | Javascript(callback or Promise) |
```

Comment:
Make the change as below.
Javascript -> JavaScript.

Reply:
Okay